File: C-PROBLEM SETS/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letter(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int number_letters = count_letter(text);
    int number_words = count_words(text);
    int number_sentences = count_sentences(text);
    float l = (number_letters / (float) number_words) * 100;
    float s = (number_sentences / (float) number_words) * 100;
    int index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");

    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letter(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            count += 1;
        }
    }
    return count;
}
int count_words(string text)
{
    int count = 1;
    for (int j = 0; j < strlen(text); j++)
    {
        if (text[j] == 32)
        {
            count += 1;
        }
    }
    return count;
}
int count_sentences(string text)
{
    int count = 0;
    for (int j = 0; j < strlen(text); j++)
    {
        if (text[j] == 33 || text[j] == 63 || text[j] == 46)
        {
            count += 1;
        }
    }
    return count;
}

File: PYTHON-SQL PROBLEM SETS/problem set 1 python/sentimental-readability/readability.py
# TODO
from cs50 import get_string
text = get_string("Text: ")
import string


num_words = 1
num_letters = 0
num_sentences = 0

for c in text:
    if c.isalpha():
        num_letters += 1
    elif c.isspace():
        num_words += 1
    elif c == "!" or c == "?" or c == ".":
        num_sentences += 1


result = round(0.0588 * (float(num_letters/num_words*100)) - 0.296 * (float(num_sentences/num_words*100)) - 15.8)

if result < 1:
    print("Before Grade 1")
elif result >= 16:
    print("Grade 16+")
else:
    print("Grade: ", result)


